### ISSUE TITLE
Valid violation not reported by redundant_type_annotation for Bool

### ISSUE THREAD

**Issue by revolter** [realm/SwiftLint#3423]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Using the `redundant_type_annotation` and `var isActive: Bool = false` doesn't trigger a warning.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ ./swiftlint lint Test.swift
Linting Swift files at paths Test.swift
Linting 'Test.swift' (1/1)
Done linting! Found 0 violations, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.41.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Using a pre-built package
* Paste your configuration file:

```yml
opt_in_rules:
    - redundant_type_annotation
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)? Xcode 12.0 Build version 12A7209
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
class Test {
	var isActive: Bool = false
}
```

**Comment by revolter** [comment]
@jpsim, Same for this one.
